source/slang/slang-ir-link.cpp

```diff
@@ -1520,6 +1521,138 @@ static void diagnoseUnresolvedSymbols(TargetRequest* req, DiagnosticSink* sink,
     }
 }
 
+void convertAtomicToStorageBuffer(
```

Comment:
Can we just expose the type as a pure user space thing? E.g. this in stdlib:

struct atomic_uint
{
RWStructuredBuffer<uint> buffer;
}

uint atomicAdd(atomic_uint v, uint x)
{
    uint oldV;
    InterlockedAdd(v.buffer[0], x, oldV);
    return oldV;
}

Reply:
we cannot due to the very unique glsl layout syntax

Reply:
in a regular structured buffer we can have an unbound int array, which is fine and works for the data storage

when fetching and indexing we would need an intrinsic much like what is being done with raytracing, this time with offset of an atomic uint; although this time todo this we would need a pass to convert all same binding atomic_uint's and field access  into 1 atomic_uint and fetch proper field access;

we would still need some magic type to work with and its still going to mangle output, we just will do it differently

Reply:
I see. Yeah this is the right thing to do.

Reply:
this as in change to the other implementation choice or keep as is (for now)?

Reply:
The current implementation is good as is.